design/baremetal-operator/bmh_boot_iso.md

```diff
@@ -0,0 +1,128 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Add boot-iso API to BareMetalHost
+
+## Status
+
+provisional
+
+## Summary
+
+Add a new interface so that it is possible to boot arbitrary
+iso images via Ironic when using redfish-virtualmedia.
+
+## Motivation
+
+In some circumstances it is desirable to boot an existing iso image
+instead of having Ironic boot IPA, for example:
+
+To reduce boot time for ephemeral workloads, it may be possible to boot an iso
+and not deploy any image to disk (saving the time to write the image and reboot)
+
+Where an alternative installer exists that is distributed as a live-iso, it
+may be desirable to leverage that toolchain instead of the IPA deploy ramdisk,
+for example the [fedora-coreos installer](https://docs.fedoraproject.org/en-US/fedora-coreos/bare-metal/)
+
+An [Ironic API](https://docs.openstack.org/ironic/latest/admin/drivers/redfish.html#virtual-media-ramdisk) exists for this, but it's not currently accessible via metal3, so it would be desirable to optionally enable this workflow.
+
+### Goals
+
+Expose the Ironic API to boot iso images via metal3.
+
+### Non-Goals
+
+This only considers the case where the redfish-virtualmedia BMC
+option is used, not any ability to boot arbitrary iso images via
```

Comment:
Ironic supports booting ISOs with iPXE as well, I think we should include it too.